### PR TITLE
Round up when calculating register size in bytes

### PR DESCRIPTION
**Pull request by Tiwalun**
This fixes the issue in #418.

### CODE REVIEW

**Comment by burrbull** [comment]
bors try

**Comment by therealprof** [comment]
Seems a bit heavy and weird to do this with floating point operations.

File: src/generate/peripheral.rs [hunk 1]
```diff
@@ -323,13 +323,13 @@ impl FieldRegions {
         let mut indices = Vec::new();
 
         let field_start = field.offset;
-        let field_end = field_start + field.size / BITS_PER_BYTE;
+        let field_end = field_start + (( field.size as f32 ) / (BITS_PER_BYTE as f32)).ceil() as u32;
```
Comment (therealprof) [inline src/generate/peripheral.rs]: The more obvious way to achieve the same would be

```suggestion
        let field_end = field_start + (field.size + BITS_PER_BYTE - 1) / BITS_PER_BYTE;
```

**Comment by Tiwalun** [comment]
You're right, it's neater this way. I applied your suggestion.

**Comment by Tiwalun** [comment]
I'm wondering if this still has an issue with larger fields, for example if I declare a 17 bit wide register,
this will use a 24 bit width to calculate the offset.

However, the register width is mapped to to the next power of two, and the register field will actually be 32 bit wide.

https://github.com/rust-embedded/svd2rust/blob/325a3044bdd2f4e8771ccd4a3a23e3a3df983de1/src/generate/register.rs#L34

It might make more sense to use the actual width of the register in this calculation, instead of calculating it again.

**Pull request merged**